profiling/src/collections/string_storage.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright 2023-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::cell::Cell;
+use std::collections::HashMap;
+use std::hash::BuildHasherDefault;
+use std::ptr;
+use std::rc::Rc;
+
+use super::identifiable::StringId;
+use super::string_table::StringTable;
+
+pub trait StringStorage {
+    /// Interns the `str` as a string, returning the id in the string table.
+    /// The empty string is guaranteed to have an id of [StringId::ZERO].
+    fn intern(&mut self, item: Rc<str>) -> StringId;
+    fn get_string(&self, id: StringId) -> Rc<str>;
+    fn len(&self) -> usize;
+    fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+    fn into_iter(self: Box<Self>) -> Box<dyn Iterator<Item = Rc<str>>>;
+    fn clone_empty(&self) -> Box<dyn StringStorage>;
+}
```

Comment:
It seems like in the current PR, this is only implemented by stuff in `experimental`. Is that intended?

Reply:
Yup, indeed none of this is needed or in use right now. In the name of this PR having less unused leftover appendages I've just gone ahead and removed it in 28782b5d2d6ec3508e234792ec94fd1ab8fa4b10 .